Sources/Basics/HTTPClient/LegacyHTTPClient.swift

```diff
@@ -358,6 +358,25 @@ extension LegacyHTTPClient {
         )
     }
 
+    public func put(
+        _ url: URL,
+        body: Data?,
+        headers: HTTPClientHeaders = .init(),
+        options: Request.Options = .init(),
+        observabilityScope: ObservabilityScope? = .none
+    ) async throws -> Response {
+        try await withCheckedThrowingContinuation { continuation in
+            self.put(
```

Comment:
I suggest having this call `execute` directly, and switching the completion handler versions to call the async versions. That way when it comes time to remove the completion handler overloads it's just a plain delete.

Reply:
Instead of building out these missing pieces in LegacyHTTPClient I switched the RegistryClient over to use the newer and asyc-ier HttpClient.